flight/utils/Wrapped_LAC.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python
+import asyncio
+from lac import *
+from mavsdk import System
+
+'''
+Given:
+  Distance = value/1024 * stroke
+  Stroke is max extension length
+  All values in mm
+  0 and 1023 are the mechanical stops **BAD**
+  Max Speed = 46 mm/s
+  Stroke = 300mm
+  
+Notes:
+  DO NOT set the rLimit or eLimit to their respective mechanical stops
+    (Unless you want to possibly break the Linear Actuator)
+  
+  Functions not used in our Actuonix Linear Actuator from 'lac.py':
+    linearActuator.set_proportional_gain()
+    linearActuator.set_derivative_gain()
+    linearActuator.disable_manual() DONT TOUCH THIS PL0x
+
+    Both of these didn't do anything when tested:
+      piston.set_average_rc(?) [0, 1024]
+      piston.set_average_adc(?) [0, 1024]
+'''
+
+'''
+Constants
+'''
+#Default IDs
+vendorID = 0x4D8
+productID= 0xFC5F
+
+
+#Retract
+rLimit = 1 #Retract limit
+rPosition = 1 #Ideal retract position
+rMechStop = 0 #Mechanical stop of retract
+
+#Extend
+eLimit = 1022 #Extend limit
+ePosition = 1022 #Ideal extend position
+eMechStop = 1023 #Mechanical stop of extend
+
+#Additional Values
+maxPWM = 1022
+minPWM = 1
+stallTime = 1000 #1000ms (1 second)
+moveThresh = 5 #Movement threshold
+accVal = 4 #Accuracy value
+maxSpeed = 1022
+sleepVal = 6 #6 seconds
+stroke = 300 #max length of LAC (mm)
+
+
+class sLAC:
+
+  def __init__(self):
+    self.piston = LAC(vendorID,productID)
+    var = self.setupLAC()
+    if (not var):
+      raise Exception("Min and Max are set to mechanical stops.")
+  
+  #Automatically sets up the LAC
+  def setupLAC(self):
+    
+    #Retract limit set to 1mm (0-1023)
+    self.piston.set_retract_limit(rLimit)
+    print("Retract limit set")
+    
+    #Extend limit set to 1022mm (0-1023)
+    self.piston.set_extend_limit(eLimit)
+    print("Extend limit set")
+    
+    #How close to target is acceptable
+    self.piston.set_accuracy(accVal)
+    print("Accuracy set")
+    
+    #Min speed (mm/s) before stalling
+    self.piston.set_movement_threshold(moveThresh)
+    print("Movement Threshold set")
+  
+    #Stall time (ms) set to 1 second
+    self.piston.set_stall_time(stallTime)
+    print("Stall Time set")
+  
+    #[1,1022]
+    self.piston.set_max_pwm_value(maxPWM)
+    print("Max PWM set")
+  
+    #[1,1022]
+    self.piston.set_min_pwm_value(minPWM)
+    print("Min PWM set")
+  
+    #Keep on max speed [1,1022]
+    self.piston.set_speed(maxSpeed)
+    print("Piston set to max speed")
+    
+    #Ensures the LAC will not hit mechanical stops
+    if (rPosition <= rMechStop):
+      return False
+    elif (ePosition >= eMechStop):
+      return False
+    else:
+      return True
```

Comment:
Move the checks for mechstop to the init function, and return nothing from this function